node-api/backend/getters.go

```diff
@@ -105,9 +106,27 @@ func (b *Backend) GetParentSlotByTimestamp(timestamp math.U64) (math.Slot, error
 	return b.sb.BlockStore().GetParentSlotByTimestamp(timestamp)
 }
 
-// GetSignatureBySlot retrieves the block signature for a given slot from the block store.
+// GetSignatureBySlot retrieves the block signature for a given slot by decoding
+// the SignedBeaconBlock from CometBFT's blockstore.
 func (b *Backend) GetSignatureBySlot(slot math.Slot) (crypto.BLSSignature, error) {
-	return b.sb.BlockStore().GetSignatureBySlot(slot)
+	block := b.node.GetBlock(int64(slot.Unwrap())) //#nosec:G115
+	if block == nil {
+		return crypto.BLSSignature{}, fmt.Errorf("block not found at slot %d", slot)
+	}
+
+	// Extract transactions from CometBFT block
+	txs := make([][]byte, len(block.Data.Txs))
+	for i, tx := range block.Data.Txs {
+		txs[i] = tx
+	}
+
+	forkVersion := b.cs.ActiveForkVersionForTimestamp(math.U64(block.Header.Time.Unix())) //#nosec:G115
```

Comment:
is this the right timestamp to use? I think it is but its technically the consensus time and not the block time